File: test/ruby/testgrpc_service.rb
require 'grpc'
require_relative '../proto/test_services_pb'
require_relative '../../src/ruby/lib/chitti'

Chitti.RpcImport(Testgrpc::TestgrpcService)

Chitti::Errors.enable([Testgrpc::CustomError], code: 600)

class TestgrpcService < Testgrpc::TestgrpcService::Service
  def hellogrpc(req, _unused_call)
    # begin
    puts "Received call in hellogrpc #{req.to_h}"
    response = Testgrpc::HelloResponse.new
    response.res_message = 'Hello from Grpc Server ! '

    return response
    c_err = Testgrpc::Errors::CustomError.new(custom: 'custom error')

    fail c_err
    # Log.log.info "response is #{response.inspect}"
    # Ticktocker::TaskFuture.new_future('dev', 'test', '123')
    # response
    # raise StandardError.new "TestError"
    #	   rescue Testgrpc::CustomError => e
    #	   	puts "coming", e.class
    #	   end
  end
end

File: test/node/testgrpc_service.js
import grpc from 'grpc';
import { RPCImport, RPCServer, StatsDInterceptor, Chitti } from '../../src/node-src/index';


const { TestgrpcService } = RPCImport(require('../proto/test.json')).testgrpc;


class MyService extends TestgrpcService.Service {
    async hellogrpc(req) {
        console.log(req.request);
        return { res_message: 'Hello Grpc !' };
    }
}

Chitti.add_handler_interceptor(StatsDInterceptor);
const grpc_server = new RPCServer();
grpc_server.addService(MyService);
grpc_server.bind('0.0.0.0:8080', grpc.ServerCredentials.createInsecure());
grpc_server.start();
